src/dist/replication/lib/replica_file_provider.cpp

```diff
@@ -20,10 +20,133 @@ error_code replica::do_download(const std::string &remote_dir,
                                 dist::block_service::block_filesystem *fs,
                                 /*out*/ uint64_t &download_file_size)
 {
-    // TODO(heyuchen): TBD
-    // download files from remote provider
-    // this function can also be used in restore
-    return ERR_OK;
+    error_code download_err = ERR_OK;
```

Comment:
I think current implementation is during a middle stage, it is okay to leave `do_download` function in replica namespace. 
If I move download function into  block_service module, I should also move unit test into block_service module. However, cold back has wrote a `block_file_mock` in replica unit test helpful for backup unit tests. If I move it into block_service, I should update backup unit tests code, which will be refactored in further pull request finally, it is not meaningful to refactor currently.
In my view, we should move all functions related to file provider into block_service module, such as download files, upload files. It can be implemented after refactoring backup and restore code.

Reply:
As we discuss offline, I will refactor code in next pull request.